### ISSUE TITLE
Request: Support class attributes

### ISSUE THREAD

**Issue by Goldziher** [snok/flake8-type-checking#128]
Hi, 

So I got a pydantic model:

```python
from pydantic import BaseConfig, BaseModel

from starlite.cache import CacheBackendProtocol
from starlite.types import CacheKeyBuilder

class CacheConfig(BaseModel):
    """Configuration for response caching.

    To enable response caching, pass an instance of this class to the
    [Starlite][starlite.app.Starlite] constructor using the
    'cache_config' key.
    """

    class Config(BaseConfig):
        arbitrary_types_allowed = True

    backend: CacheBackendProtocol
    expiration: int = 60 
    cache_key_builder: CacheKeyBuilder
```

Both the `backend` and `cache_key_builder` attributes are typed using custom classes, which are imported only for typing purposes. Hence this would be possible:

```python
from typing import TYPE_CHECKING

from pydantic import BaseConfig, BaseModel

if TYPE_CHECKING:
    from starlite.cache import CacheBackendProtocol
    from starlite.types import CacheKeyBuilder

class CacheConfig(BaseModel):
    """Configuration for response caching.

    To enable response caching, pass an instance of this class to the
    [Starlite][starlite.app.Starlite] constructor using the
    'cache_config' key.
    """

    class Config(BaseConfig):
        arbitrary_types_allowed = True

    backend: "CacheBackendProtocol"
    expiration: int = 60 
    cache_key_builder: "CacheKeyBuilder"
```

I would like to ask for this feature to be included in this plugin.

**Comment by sondrelg** [comment]
Running flake8 on the bottom example produces no TC errors when testing. What exactly do you mean when you say include the feature?

If you mean that imports used type hinting should be flagged to have them moved *into* type-checking blocks, then I don't think that's possible, since last time I checked Pydantic evaluates the type hints at runtime. Scoping them in runtime-exempt code paths (which is what type checking blocks are) would lead to missing reference (`NameError`s), no?

**Comment by Goldziher** [comment]
> Running flake8 on the bottom example produces no TC errors when testing. What exactly do you mean when you say include the feature?
> 
> If you mean that imports used type hinting should be flagged to have them moved *into* type-checking blocks, then I don't think that's possible, since last time I checked Pydantic evaluates the type hints at runtime. Scoping them in runtime-exempt code paths (which is what type checking blocks are) would lead to missing reference errors (`NameError`s), no?

Pydantic resolves these using the method `resolve_foward_refs`.

It could be that for pydantic this shouldn't be an error, but what about regular classes?

**Comment by sondrelg** [comment]
The problem is that we cannot know whether a class is a Pydantic class or regular class, if it has any base model. We're constrained to the scope of the individual file we're looking at, and the base class could be defined outside that scope. 

tl;dr: if pydantic support is enabled, we treat any class with a base class as a *potential* pydantic class. More context in #58

**Comment by Goldziher** [comment]
well, you can expose a config option "pydantic_base_classes" or simply "exclude_for_attributes_check"

**Comment by sondrelg** [comment]
The exclude for attributes check option exists [here](https://github.com/snok/flake8-type-checking#pydantic-support-base-class-passlist) 👍

**Comment by Goldziher** [comment]
I would reverse the config though - instead of a "pass list", an "pydantic include list", and all other classes shouldnt be considered pydantic.
